install_files/securedrop-keyring/DEBIAN/preinst

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+set -e
+
+# Solution adapted from DKG's work on `deb.torproject.org-keyring`.
+# In SecureDrop versions before 0.3.10, the Ansible playbooks used
+# the apt_key module to add the signing key to /etc/apt/trusted.gpg.
+# It's cleaner to use the trusted.gpg.d subdirectory, since we can
+# update that trivially in future versions of the keyring package.
+#
+# Therefore let's clean up prior versions of the key installed
+# to the general apt keyring, to ensure we only have one signing key
+# installed for authenticating securedrop-related packages.
+
+if [ -e /etc/apt/trusted.gpg ] && which gpg >/dev/null; then
+   (
+   h="$(mktemp -d)"
+   trap "rm -rf '$h'" EXIT
+
+   if gpg --homedir="$h" \
+          --batch --no-tty --no-default-keyring --keyring /etc/apt/trusted.gpg \
+          --list-key 0xB89A29DB2128160B8E4B1B4CBADDE0C7FC9F6818 > /dev/null 2>&1 ; then
+      gpg --homedir="$h" \
+          --batch --no-tty --no-default-keyring --keyring /etc/apt/trusted.gpg \
+          --no-auto-check-trustdb \
+          --delete-key 0xB89A29DB2128160B8E4B1B4CBADDE0C7FC9F6818 || true
```

Comment:
Struck me as strange as well, but I decided to reuse DKG's approach. A bit cargo-culty, but it definitely works as expected. Going forward, as we modify the `preinst` script to accommodate future rotations or other features, we can make the script more readable.

What I'd really like is the ability to reuse the keyid from an Ansible variable, so it's DRY across the entire project—that will take more consideration to do sanely, though.